lib/puppet-editor-services/json_messages.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+require 'puppet-editor-services/json_rpc_handler'
+
+module PuppetEditorServices
+  module JSONMessage
+    def self.reply_result(raw_request, result)
+      {
+        KEY_JSONRPC => VALUE_VERSION,
+        KEY_ID      => raw_request[KEY_ID],
+        KEY_RESULT  => result
+      }
+    end
+
+    def self.reply_error(raw_request, code, message)
+      {
+        KEY_JSONRPC => VALUE_VERSION,
+        KEY_ID      => raw_request[KEY_ID],
+        KEY_ERROR   => {
+          KEY_CODE    => code,
+          KEY_MESSAGE => message
+        }
+      }
+    end
+
+    def self.reply_method_not_found(raw_request, message = nil)
+      reply_error(raw_request, CODE_METHOD_NOT_FOUND, message || MSG_METHOD_NOT_FOUND)
+    end
+
+    def self.notify_diagnostics(uri, diagnostics)
```

Comment:
This specific one should NOT be here.  sending a notification is find, but that's a hardcoded LSP thing. Not common between LSP and DSP